[RFC] ceph iscsi config: add max_data_area setting

mikechristie opened the pull request:
Allow users to config the tcmu data area ring size.

This reqiures a updated kernel with the patch:

https://www.spinics.net/lists/target-devel/msg16097.html

and a updated rtslib with this PR merged:

https://github.com/open-iscsi/rtslib-fb/pull/112

This is marked a RFC because we are still waiting on upstream testing/review, so the patch should not be merged yet.

dillaman reviewed (commented):
Failing to set ```max_data_area_mb``` doesn't seem like it should be a fatal error to me since it doesn't impact correctness or safety. It seems like it would just be better to warn if that value cannot be set.

mikechristie commented:
Ok. Will fix.

dillaman commented:
@mikechristie alternative PR #32

Commit:
ceph iscsi config: add max_data_area setting

Allow users to config the tcmu data area ring size.

This reqiures a updated kernel with the patch:

://www.spinics.net/lists/target-devel/msg16140.htmlttps://www.spinics.net/lists/target-devel/msg16097.html

and a updated rtslib with this PR merged:

open-iscsi/rtslib-fb#112
Signed-off-by: Mike Christie <mchristi@redhat.com>

dillaman commented:
lgtm -- pending rtslib_fb change

mikechristie commented:
Dropping this PR. Following the idea here

https://github.com/open-iscsi/targetcli-fb/pull/95

where we do not want to update every tool/lib in the stack to add a setting, we can modify rtslib-fb and our tools/libs to take control settings after the initial object initialization but before enablement, and we can do the same for ceph-iscsi-config/gwcli.

Will make new PRs.
